pkg/proxy/proxy.go

```diff
@@ -349,7 +349,26 @@ func BootProxy(logger *zap.Logger, opt Option, appCmd, appContainer string, pid
 	if opt.Port == 0 {
 		opt.Port = 16789
 	}
+	maxAttempts := 1000
+	attemptsDone:=0
 
+
+    if !isPortAvailable(opt.Port) {
+		for i := 1024; i <= 65535 && attemptsDone < maxAttempts; i++ {
+			if isPortAvailable(uint32(i)) {
+				opt.Port = uint32(i)
+				logger.Info(Emoji+"Port %v is already in use, switching to next available port");
+
+				break
+			}
+			attemptsDone++
+		}
+	}
+	
+	if maxAttempts<=attemptsDone{
```

Comment:
beautify the statement I guess there must be spaces before and after the <= symbol. There must be some extension in the IDE you can use it and make sure it is indentation is correct.